lib/puppet/provider/service/service.rb

```diff
@@ -36,6 +36,9 @@ def ucommand(type, fof = true)
     else
       cmd = [send("#{type}cmd")].flatten
     end
+    if (( type =~ /^start/ ) && ( Facter.value(:osfamily) == "Solaris" ))
+      cmd = ["/usr/bin/ctrun -l none", cmd].flatten.join(" ")
+    end
```

Comment:
@joshcooper Thanks, that approach makes more sense.

I'm not sure if I'm asking the right question, but will this work to override a method in a "grandparent" class? (The ```texecute``` method is in ```service```, which is the parent of ```base```, which is the parent of ```smf```.)